Technical-Steering-Committee/best-practices/build-and-test.md

```diff
@@ -0,0 +1,17 @@
+# Build and Test
+
+## Build
+If a Zowe component or a repository requires source code compilation, this repository should contain automated build system able to compile the software from source, automatically handle dependencies and produce installation packages. The build tools in use should be only FLOSS, free to acquire and use for any purposes. It is recommended to select tools already popular within Zowe, such as Gradle, npm, GitHub Actions, etc. For specific needs, it is possible to select alternatives, given they provide unique and necessary advantages.
+
+## Test
+Every Zowe component or repository should contain an automated test suit that is executed on every change or code build. Ideally the automated test suit should consist of the following categories of tests - unit tests, integration tests and system tests. The tests should cover both functional, i.e. validating a feature functionality, as well as non-functional, i.e. performance, reliability, resilience, etc. 
```

Comment:
May be the performance, reliability and resilience testing should be stated as a recommendation only.

The previous sentence we says:  ."..repository SHOULD contain automated tests, which .... SHOULD consist of ..." and then "The  tests SHOULD cover ..... as well performance, .....".

What I mean is, that the first and second SHOULD are like hard requirements or a MUST, while the second SHOULD is a recommendation.
 
The comment is about the wording. No discussion that it is a best practice to cover also non-functional aspects of the SW.

Reply:
This is a best practices document, it is not a condition for a repository being part of Zowe. I wouldn't change the wording. Maybe we could ask for a second opinion? @balhar-jakub

Reply:
@boris-bc While we are talking about Best practices, the following of them is actually required for the squads. So we should keep this in mind.

Reply:
OK, so what change would you suggest to the text in question?